domain/credential/state/state.go

```diff
@@ -71,101 +63,90 @@ func (st *State) credentialUUIDForKey(
+AND    owner_name = $credentialKey.owner_name
+AND    cloud_name = $credentialKey.cloud_name
+`, dbKey, credentialUUID{})
 	if err != nil {
 		return "", errors.Capture(err)
 	}
-	uuid := sqlair.M{}
-	err = tx.Query(ctx, selectStmt, credentialKeyMap(key)).Get(&uuid)
+
+	result := credentialUUID{}
+	err = tx.Query(ctx, selectStmt, dbKey).Get(&result)
 	if errors.Is(err, sql.ErrNoRows) {
 		return "", errors.Errorf("cloud credential %q %w", key, credentialerrors.NotFound)
 	} else if err != nil {
 		return "", errors.Errorf("fetching cloud credential %q: %w", key, err)
 	}
-
-	uuidStr, exists := uuid["uuid"]
-	if !exists {
-		return "", errors.Errorf(
-			"%w expected cloud credential uuid for credential %q, got no returned value",
-			credentialerrors.NotFound, key)
-
-	}
-
-	return corecredential.UUID(uuidStr.(string)), nil
+	return corecredential.UUID(result.UUID), nil
 }
 
 // UpsertCloudCredential adds or updates a cloud credential with the given name,
-// cloud and owner. If the credential exists already, the existing credential's
-// Invalid value is returned.
+// cloud and owner.
 //
 // If the owner of the credential can't be found then an error satisfying
 // [usererrors.NotFound] will be returned.
-func (st *State) UpsertCloudCredential(ctx context.Context, key corecredential.Key, credential credential.CloudCredentialInfo) (*bool, error) {
+func (st *State) UpsertCloudCredential(ctx context.Context, key corecredential.Key, credential credential.CloudCredentialInfo) error {
 	db, err := st.DB()
 	if err != nil {
-		return nil, errors.Capture(err)
+		return errors.Capture(err)
 	}
 
-	q := `
-SELECT  uuid AS &M.uuid,
-        invalid AS &M.invalid
-FROM v_cloud_credential
-WHERE cloud_name = $M.cloud_name
-AND name = $M.credential_name
-AND owner_name = $M.owner
-`
-	stmt, err := st.Prepare(q, sqlair.M{})
+	dbKey := credentialKey{
+		CredentialName: key.Name,
+		CloudName:      key.Cloud,
+		OwnerName:      key.Owner.String(),
+	}
+	stmt, err := st.Prepare(`
+SELECT uuid AS &credentialUUID.uuid
+FROM   v_cloud_credential
+WHERE  name = $credentialKey.name
+AND    owner_name = $credentialKey.owner_name
+AND    cloud_name = $credentialKey.cloud_name
+`, dbKey, credentialUUID{})
 	if err != nil {
-		return nil, errors.Capture(err)
+		return errors.Capture(err)
 	}
 
-	var existingInvalid *bool
 	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
 		// Get the credential UUID - either existing or make a new one.
 		// TODO(wallyworld) - implement owner as a FK when users are modelled.
 
-		result := sqlair.M{}
-		err = tx.Query(ctx, stmt, credentialKeyMap(key)).Get(result)
+		result := credentialUUID{}
+		err = tx.Query(ctx, stmt, dbKey).Get(&result)
 		if err != nil && !errors.Is(err, sqlair.ErrNoRows) {
 			return errors.Capture(err)
 		}
-		invalid, ok := result["invalid"].(bool)
-		if ok {
-			existingInvalid = &invalid
-		}
-		credentialUUID, ok := result["uuid"].(string)
-		if !ok {
+		if err != nil {
```

Comment:
This should be an else if for `errors.Is(err, sqlair.ErrNoRows)`

Reply:
I was trying to avoid nesting

Reply:
This should still check for the no rows err and not assume.